frontend/src/container/ListAlertRules/index.tsx

```diff
@@ -1,22 +1,45 @@
+import { notification } from 'antd';
 import getAll from 'api/alerts/getAll';
 import Spinner from 'components/Spinner';
-import React from 'react';
+import React, { useEffect } from 'react';
 import { useTranslation } from 'react-i18next';
 import { useQuery } from 'react-query';
 
 import ListAlert from './ListAlert';
 
 function ListAlertRules(): JSX.Element {
 	const { t } = useTranslation('common');
-	const { data, isError, isLoading, refetch } = useQuery('allAlerts', {
+	const { data, isError, isLoading, refetch, status } = useQuery('allAlerts', {
 		queryFn: getAll,
 		cacheTime: 0,
 	});
 
+	useEffect(() => {
+		if (status === 'error' || (status === 'success' && data.statusCode !== 200)) {
+			notification.error({
+				message: data?.error || t('something_went_wrong'),
+			});
+		}
+	}, [data?.error, data?.statusCode, status, t]);
+
+	// api failed to load the data
 	if (isError) {
 		return <div>{data?.error || t('something_went_wrong')}</div>;
 	}
 
+	// api is successful but error is present
+	if (status === 'success' && data.statusCode !== 200) {
```

Comment:
is status 200 is the correct approach because we might have apis return 201 or 200
Can we check for series of 200 or statusCode < 400

Reply:
here we only requesting the server to fetch the details ... not creating the data (201).. also when the data is successfully fetched we return 200

Reply:
Just to be on the safer side I have seen people using this check
Error if status code >= 400
Not error < 400

Reply:
this is already handled in the Error-Handler _(with Axolo)_